src/cmps/tags/Tags.tsx

```diff
@@ -1,51 +1,92 @@
-import React, { Component } from "react";
+import React, { Component } from 'react';
 import './Tags.css';
 import Tag from '../tag/Tag';
-// eslint-disable-next-line
-import { string } from "prop-types";
+// import { string, number } from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { library } from '@fortawesome/fontawesome-svg-core';
+import { faPlus } from '@fortawesome/free-solid-svg-icons';
+// import { any, element } from 'prop-types';
 
+library.add(faPlus);
 
 class Tags extends Component {
-
     state = {
-        tags: [{
-            id: 1,
-            name: 'tag1'
-        }, {
-            id: 2,
-            name: 'tag2'
-        }]
-    }
+        currentTag: '',
+        tags: [],
+    };
 
-    remove = () => {
-        console.log('button X clicked');
-    }
+    remove = (id: string) => {
+        let index = this.state.tags.findIndex((element: any) => element.id === id);
+        this.setState(this.state.tags.splice(index, 1));
+    };
+
+    handleSubmit = (e: any) => {
+        e.preventDefault();
+        let isDouble = this.checkDouble(this.state.currentTag);
+        if (isDouble) {
+            let title = this.state.currentTag;
+            let id = this.makeId(9);
+            if (title.length <= 0) return;
+            this.setState({
+                currentTag: '',
+                tags: [...this.state.tags, { title, id }],
```

Comment:
important 🚩
same comment as before
(State Updates May Be Asynchronous)

Reply:
same comment as before...